qutip_lattice/lattice.py

```diff
@@ -200,1108 +257,6196 @@ def cell_structures(val_s=None, val_t=None, val_u=None):
+    symOpInvariantsUp = np.array([np.power(1, np.arange(latticeSize[0]))])
+
+    kValue = kVector[kval]
+    basisStates_S = createHeisenbergfullBasis(latticeSize[0])
+
+    [DownStatesPerk, Ind2kDown, normDownHubbardStatesk
+     ] = combine2HubbardBasisOnlyTrans(symOpInvariantsUp, basisStates_S,
+                                        latticeSize, kValue, PP, Nmax=1)
+    compDownStatesPerRepr_S = DownStatesPerk
+    compInd2ReprDown_S = Ind2kDown
+    symOpInvariantsUp_S = np.array([np.ones(latticeSize[0], )])
+    normHubbardStates_S = normDownHubbardStatesk
+    [indNeighbors, nSites] = getNearestNeighbors(
+        latticeType=latticeType, latticeSize=latticeSize,
+        boundaryCondition=PBC_x)
+    H_down = calcHamiltonDownOnlyTrans(
+        compDownStatesPerRepr_S, compInd2ReprDown_S, paramT, indNeighbors,
+        normHubbardStates_S, symOpInvariantsUp_S, kValue, basisStates_S,
+        latticeSize, PP)
+
+    return [Qobj(H_down), compDownStatesPerRepr_S]
+
+
+def spinlessFermions_NoSymHamiltonian(paramT, latticeType, latticeSize, PP,
+                                       PBC_x):
+    """
+    Calculates the Hamiltonian for a no symmetry specified basis.
+
+    Parameters
+    ==========
+    paramT : float
+        The nearest neighbor hopping integral
+    latticeType : str
+        latticeType = 'cubic' indicates specification on a cubic lattice
+    latticeSize : list of int
+        it has a single element as the number of cells as an integer.
+    PP : int
+        The exchange phase factor for particles, +1 for bosons, -1 for fermions
+    PBC_x : int
+        indication (1) of specification of periodic boundary condition or (0)
+        open boundary condition.
+
+    Returns
+    -------
+    Qobj(H_down) : Qobj(csr_matrix)
+        The matrix elements that are associated with hopping between bosons on
+        nearest neighbor sites.
+    compDownStatesPerRepr_S : numpy 2d array
+        Each row indicates a basis vector chosen in the number and k-vector
+        symmetric basis.
+    """
+    nSites = latticeSize[0]
+    kVector = np.arange(start=0, stop=2 * np.pi, step=2 * np.pi / nSites)
+    symOpInvariantsUp = np.array([np.zeros(latticeSize[0], )])
+    symOpInvariantsUp[0, 0] = 1
+    kval = 0
+    kValue = kVector[kval]
+    [indNeighbors, nSites] = getNearestNeighbors(
+        latticeType=latticeType, latticeSize=latticeSize,
+        boundaryCondition=PBC_x)
+
+    basisStates_S = createHeisenbergfullBasis(latticeSize[0])
+    [compDownStatesPerRepr, compInd2ReprDown, normHubbardStates
+     ] = combine2HubbardBasisOnlyTrans(symOpInvariantsUp, basisStates_S,
+                                        latticeSize, kValue, PP, Nmax=1)
+    H_down = calcHamiltonDownOnlyTrans(
+        compDownStatesPerRepr, compInd2ReprDown, paramT, indNeighbors,
+        normHubbardStates, symOpInvariantsUp, kValue, basisStates_S,
+        latticeSize, PP)
+
+    return [Qobj(H_down), basisStates_S]
```

Comment:
What are the dimension of the Hamiltonian? They should reflect the lattice structure.
Also can this Hamiltonian be non-hermitian? If not, it should be flagged as such.